libminifi/src/io/StreamFactory.cpp

```diff
@@ -38,35 +39,21 @@ namespace io {
 
 template<typename T, typename V>
 class SocketCreator : public AbstractStreamFactory {
-  template<bool cond, typename U>
```

Comment:
I understand that these were noops in this form, but could you decipher the original intent? (Not the implementation reality.)

Reply:
No, I couldn't decipher the intent

Reply:
first introduction: https://github.com/apache/nifi-minifi-cpp/commit/44704b363ccc422bb4fe2e1a6dcc184e9213bd68#diff-d6d59c6c5b3c0aac82860661f0e4183fR43

The intent is still unclear. Maybe it was meant to be a SFINAE check whether `T` can be instantiated like a `Socket` can, and if not, then fall back to `Socket`. I'm not sure, this is just a wild guess. We will get a compilation error anyway if it's ever used like that.